pallets/pool-registry/src/lib.rs

```diff
@@ -387,12 +401,63 @@ pub mod pallet {
 				},
 			);
 
-			Self::deposit_event(Event::MetadataSet {
-				pool_id,
-				metadata: checked_metadata,
-			});
+			Ok(checked_metadata)
+		}
+	}
 
-			Ok(())
+	impl<T: Config> cfg_traits::PoolMetadata<T::Balance, VersionedMultiLocation> for Pallet<T> {
```

Comment:
What do you think about moving this to a `impls.rs` mod?

Reply:
Not a blocker for me btw

Reply:
Since we don't have an `impls.rs` mod yet, I would leave it here. Other pallets also have such impls in the `lib.rs`. I agree, all of that should probably be moved to impls. So it can be touched all at once in a cleanup PR later this year.